### PR TITLE
Document Inheritance, Connection Pool, MySQL max_prepared_stmt_count issue, MSSQL JSON 

### PR DESCRIPTION
**Pull request by ephys**
This PR documents:

- https://github.com/sequelize/sequelize/pull/15832
- https://github.com/sequelize/sequelize/pull/16095
- How to configure the pool properly
- How to monitor the pool
- What to do when you get a `ConnectionAcquireTimeoutError` (we'll need to make the error point to this page)
- What to do if you get a "Can't create more than max_prepared_stmt_count statements" error

This PR moves:

- Optimistic locking to the model category (future PR will rewrite the page to provide more information)
- Virtual attributes to the model category

### CODE REVIEW

File: docs/models/model-typing.mdx [hunk 1]
```diff
@@ -75,14 +75,6 @@ You only need to use `CreationOptional` & `NonAttribute` on class instance field
 
 Example of a minimal TypeScript project with strict type-checking for attributes:
 
-import CodeBlock from '@theme/CodeBlock';
-import modelInitExample from '!!raw-loader!@site/.sequelize/v7/packages/core/test/types/typescript-docs/model-init.ts';
-
-<CodeBlock className="language-typescript">
```
Comment (ephys) [inline docs/models/model-typing.mdx:81]: This complete example felt really noisy and redundant now that the rest of the documentation is typescript-first

File: docs/models/concrete-table-inheritance.svg
Comment (WikiRik) [inline docs/models/concrete-table-inheritance.svg:1]: There's two typo's in the bowlers table.
`Name` should be `name` I think
`battling_average` should be `batting_average`

File: docs/models/inheritance.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+class Footballer extends Player<Footballer> {
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  declare club: string;
+}
+
+class Cricketer<M extends Cricketer = Cricketer> extends Player<M> {
+  @Attribute(DataTypes.INTEGER)
+  @NotNull
+  declare battlingAverage: number;
```
Comment (WikiRik) [inline docs/models/inheritance.md:41]:
```suggestion
  declare battingAverage: number;
```

File: docs/other-topics/faq.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+1. Determine the value of `max_prepared_stmt_count` on your MySQL server:
+   ```sql
+   SHOW VARIABLES LIKE 'max_prepared_stmt_count';
+   ```
+2. Determine how many concurrent connections can be connected to your database at the same time.
+   ```sql
+   SHOW VARIABLES LIKE 'max_connections';
+   ```
+3. Set `maxPreparedStatements` to the result of `FLOOR(max_prepared_stmt_count / max_connections)`.
```
Comment (WikiRik) [inline docs/other-topics/faq.md:52]: Is this something we can add as a new feature? That sequelize automatically sets it with the initial connection?

Comment (ephys) [reply docs/other-topics/faq.md:52]: We could, and there is a thread open about it, but we need to be careful about how we do it. We could introduce issues if the dev massively increased the connection limit without changing the statement limit.

Something we could definitely do is catch the error thrown by mysql2 to append a link to this documentation

File: docs/models/concrete-table-inheritance.svg [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      <rect stroke-linecap="butt" x="625" y="483" transform="matrix(1,0,0,1,-617,-475)" clip-path="url(#clipPath2)" fill="none" width="115" image-rendering="auto" rx="4" ry="4" height="107" stroke-miterlimit="1.45"/>
+      <line stroke-linecap="butt" transform="matrix(1,0,0,1,-617,-475)" clip-path="url(#clipPath2)" fill="none" x1="625" x2="740" y1="524.4023" image-rendering="auto" y2="524.4023" stroke-miterlimit="1.45"/>
+    </g>
+    <g stroke-miterlimit="1.45" font-family="sans-serif" transform="matrix(1,0,0,1,-185,-209)" stroke-linecap="butt">
+      <text x="662.8154" xml:space="preserve" y="501.0645" clip-path="url(#clipPath2)" stroke="none">«table»</text>
+      <text x="661.1602" xml:space="preserve" y="515.7656" clip-path="url(#clipPath2)" stroke="none">bowlers</text>
+      <text x="629" xml:space="preserve" y="542.4668" clip-path="url(#clipPath2)" stroke="none">Name</text>
+      <text x="629" xml:space="preserve" y="557.168" clip-path="url(#clipPath2)" stroke="none">battling_average</text>
```
[the remark this answers is not in the document]
Comment (ephys) [reply docs/models/concrete-table-inheritance.svg]: Fixed

File: docs/models/inheritance.md (hunk 2 above)
Comment (ephys) [reply docs/models/inheritance.md]: Fixed

**Review by WikiRik — APPROVED** [review]

File: docs/models/model-typing.mdx (hunk 1 above)
Comment (ephys) [reply docs/models/model-typing.mdx:81]: https://github.com/sequelize/sequelize/pull/16194/files

File: docs/other-topics/faq.md (hunk 3 above)
Comment (ephys) [reply docs/other-topics/faq.md:52]: Let's continue this discussion in https://github.com/sequelize/sequelize/issues/10832

**Pull request merged**